crates/opencascade/examples/chamfer.rs

```diff
@@ -0,0 +1,47 @@
+use glam::dvec3;
+use opencascade::{
+    primitives::{Edge, Face, Solid, ToAngle},
+    workplane::Workplane,
+};
+
+pub fn main() {
+    let mut base = Workplane::xy().rect(10.0, 10.0);
+    // asymmetric chamfer
+    base.chamfer(1.0, Some(2.0));
+
+    let mut top = Workplane::xy().rect(10.0, 10.0);
+    // 45 degree chamfer - other angles will be asymmetric
+    top.chamfer_angle(2.0, 45.degrees());
+    top.translate(dvec3(0.0, 0.0, 10.0));
+
+    let chamfered_box = Solid::loft([&base, &top].into_iter());
+
+    let handle = Workplane::xy().rect(5.0, 5.0);
+    let handle_face = Face::from_wire(&handle);
+
+    let handle_body = handle_face.extrude(dvec3(0.0, 0.0, -10.0));
+    let mut chamfered_shape = chamfered_box.union(&handle_body);
+
+    // chamfer the top of the protrusion by getting shape edges at z = -10.0
+    let top_edges = chamfered_shape
+        .edges()
+        .filter(|edge| edge.start_point().z == -10.0 && edge.end_point().z == -10.0)
+        .collect::<Vec<Edge>>();
+    chamfered_shape.chamfer_edges(1.0, &top_edges);
+
+    // chamfer the handle join edges
+    let handle_join_edges = chamfered_shape
+        .edges()
+        .filter(|edge| {
+            let start = edge.start_point();
+            let end = edge.end_point();
+            (start.x.abs() == 2.5 && start.z == 0.0) || (end.y.abs() == 2.5 && end.z == 0.0)
+        })
```

Comment:
Similar situation here, if we continued with this approach, we should definitely move the `5.0` into a variable (let's call it `handle_thickness`), and then use `handle_thickness / 2.0` here so we don't have to update this code if that thickness changes. We would also want to use the threshold concept I mentioned above.

_But_, OpenCascade should actually be able to give us the list of edges that were generated when running a boolean operation. You'll notice the `subtract` function actually does this already. I tried to use that same concept for the `union` operation but for some reason it's giving me zero edges. Maybe I was doing something wrong though, I'm not sure.

If you want to play around with it a bit though, it would certainly make this snippet of code more concise and straightforward.

Here's a C++ example where a Fuse operation is performed and then the newly created edges are fetched:

https://neweopencascade.wordpress.com/2018/10/17/lets-talk-about-fillets/